internal/controller/proxmoxcluster_controller.go

```diff
@@ -233,13 +259,91 @@ func (r *ProxmoxClusterReconciler) listProxmoxMachinesForCluster(ctx context.Con
 	return machineList.Items, nil
 }
 
-// SetupWithManager sets up the controller with the Manager.
-func (r *ProxmoxClusterReconciler) SetupWithManager(ctx context.Context, mgr ctrl.Manager) error {
-	return ctrl.NewControllerManagedBy(mgr).
-		For(&infrav1alpha1.ProxmoxCluster{}).
-		WithEventFilter(predicates.ResourceNotPaused(ctrl.LoggerFrom(ctx))).
-		Watches(&clusterv1.Cluster{},
-			handler.EnqueueRequestsFromMapFunc(util.ClusterToInfrastructureMapFunc(ctx, infrav1alpha1.GroupVersion.WithKind(infrav1alpha1.ProxmoxClusterKind), mgr.GetClient(), &infrav1alpha1.ProxmoxCluster{})),
-			builder.WithPredicates(predicates.ClusterUnpaused(ctrl.LoggerFrom(ctx)))).
-		Complete(r)
+func (r *ProxmoxClusterReconciler) reconcileNormalCredentialsSecret(ctx context.Context, clusterScope *scope.ClusterScope) error {
+	proxmoxCluster := clusterScope.ProxmoxCluster
+	if !hasCredentialsRef(proxmoxCluster) {
+		return nil
+	}
+
+	secret := &corev1.Secret{}
+	secretKey := client.ObjectKey{
+		Namespace: proxmoxCluster.Namespace,
+		Name:      proxmoxCluster.Spec.CredentialsRef.Name,
+	}
+	err := r.Client.Get(ctx, secretKey, secret)
+	if err != nil {
+		return err
+	}
+
+	helper, err := patch.NewHelper(secret, r.Client)
+	if err != nil {
+		return err
+	}
+
+	// Ensure the ProxmoxCluster is an owner and that the APIVersion is up-to-date.
+	secret.SetOwnerReferences(clusterutil.EnsureOwnerRef(secret.GetOwnerReferences(),
+		metav1.OwnerReference{
+			APIVersion: infrav1alpha1.GroupVersion.String(),
+			Kind:       "ProxmoxCluster",
```

Comment:
use resource.Kind

Reply:
Yes, thanks!

Reply:
Well... That doesn't seem to work strangely. If I create an object with the correct "TypeMeta" (i.e. Kind, etc). Then a "Create" will remove the GVK information. Not sure why that is.

Reply:
yes